Objective-C/CBLCoreMLPredictiveModel.m

```diff
@@ -0,0 +1,538 @@
+//
+//  CBLCoreMLPredictiveModel.m
+//  CouchbaseLite
+//
+//  Copyright (c) 2019 Couchbase, Inc. All rights reserved.
+//
+//  Licensed under the Couchbase License Agreement (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//  https://info.couchbase.com/rs/302-GJY-034/images/2017-10-30_License_Agreement.pdf
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+#import "CBLCoreMLPredictiveModel.h"
+#import "CBLCoreMLPredictiveModel+Internal.h"
+#import "CBLBlob.h"
+#import "CBLDatabase+Prediction.h"
+#import "CBLDictionary.h"
+#import "CBLMutableArray.h"
+#import "CBLMutableDictionary.h"
+
+#import <CoreImage/CoreImage.h>
+#import <Vision/Vision.h>
+
+#define kCBLMayUseVisionModel YES
+#define kCBLMaxVisionPredictedProbabilities 10
+
+@implementation CBLCoreMLPredictiveModel {
+    MLModel* _model;
+    VNCoreMLModel* _vnModel;
+    NSString* _vnFeatureName;
+}
+
+- (instancetype) initWithMLModel: (MLModel*)model {
+    self = [super init];
```

Comment:
```suggestion
    CBLAssertNotNil(model);
    self = [super init];
```

Reply:
Right. We should validate non-null as well. I will fix that.